scene/gui/text_edit.h

```diff
@@ -37,6 +37,15 @@
 #include "scene/main/timer.h"
 #include "scene/resources/syntax_highlighter.h"
 #include "scene/resources/text_paragraph.h"
+#include "core/object/object.h"
+
+class TextEditSearchResult : public Object {
+	GDCLASS(TextEditSearchResult, Object);
+
+public:
+	Point2i point;
+	int match_i;
+};
```

Comment:
```suggestion
struct TextEditSearchResult {
	Point2i point;
	int match_i;
};
```

Could this be made into a struct? I think that would solve the memory leak issues too (I'm still a C++ noob, but that's the impression I've gotten from other structs like String, Vector2, etc.)

If you do end up changing it to a struct, you'll have to change all usages of `->` to `.` for TextEditSearchResult.

Reply:
If it's bound to the GDScript, it should inherit `Reference`, be registered in ClassDB and use `Ref<Type>` "smart pointer", but it's overkill for the simple object like this, `Dictionary` is probably better.

For the internal method, it might be better to use arguments passed by reference to return values and do not bother with the struct and memory management.

```c++
void TextEdit::search(const String &p_key, uint32_t p_search_flags, int p_from_line, int p_from_column, Vector2 &r_point, int &r_match_i) const
```

Reply:
I still not sure why we need to touch `TextEdit` at all? Can we not keep a count inside `_update_results_count`. If the pos is the same as our search then we have the match number.